app/src/components/GameLore.tsx

```diff
@@ -62,6 +66,40 @@ export default function GameLore(props: GameLoreProps) {
     });
   };
 
+  const handleFilterAgents = (uuids: string[]) => {
+    const newFilteredAgents = props.agents.filter(agent =>
+      uuids.includes(agent.uuid!),
+    );
+
+    setFilteredAgents(newFilteredAgents);
+  };
+
+  const showSharedLore = useCallback((lore: Lore, agents: AgentDef[]) => {
+    return agents
+      .map(agent => agent.uuid)
+      .every(element => lore.known_by?.includes(element!));
+  }, []);
+
+  const setLoreIndices = useCallback(
```

Comment:
Maybe put this login in the useEffect instead of a useCallback? Only used in one place and it would get rid of some lines of code.